ci-operator/step-registry/workers-rhel/ibmcloud-provision/workers-rhel-ibmcloud-provision-commands.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+  export IBMCLOUD_CLI=ibmcloud
+  export IBMCLOUD_HOME=/output
+  region="${LEASED_RESOURCE}"
+  export region
+  echo "Try to login..."
+  "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+ibmcloud_login
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+export SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+export PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+
+# Get existing vnet info
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ ! -f "${VPC_CONFIG}" ]]; then
+  echo "Fail to find of VPC info file ${VPC_CONFIG}, abort." && exit 1
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+infra_id=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+${IBMCLOUD_CLI} target -g ${resource_group}
+
+workdir=`mktemp -d`
+
+#when run more jobs at the same time, fail to create key with the same fingerprint, so create the key (ci-qe-key) in Default group for all test. 
+${IBMCLOUD_CLI} is keys --all-resource-groups
+
+#get the ssh key name from vault
+rhelSSHKey="$(cat ${CLUSTER_PROFILE_DIR}/ibmcloud-sshkey-name)"
+
+#use the same sgs created by installer, shared with the default rhcos nodes
+rhel_worker_sgs="${infra_id}-sg-cluster-wide,${infra_id}-sg-openshift-net"
+
+# Start to provision rhel instances from template in existing VPC and NSG
+for count in $(seq 1 ${RHEL_WORKER_COUNT}); do
+  echo "$(date -u --rfc-3339=seconds) - Provision ${infra_id}-rhel-${count} ..."
+  # Get computeSubnet
+  idx=$(((count-1) % 3))
```

Comment:
I think we should not use `3` hard code here, but it is the length of subsets from the vpc.